src/reason-parser/reason_parser.mly

```diff
@@ -3330,54 +3352,23 @@ expr_list:
 
 (* [x, y, z, ...n] --> ([x,y,z], Some n) *)
 expr_comma_seq_extension:
-  lseparated_nonempty_list(COMMA, expr_seq_item) COMMA?
+  lseparated_nonempty_list(COMMA, opt_spread(expr_optional_constraint)) COMMA?
   { match List.rev $1 with
     (* Check if the last expr has been spread with `...` *)
     | ((dotdotdot, e) as hd)::es ->
       let (es, ext) = match dotdotdot with
       | Some dotdotdotLoc -> (es, Some e)
       | None -> (hd::es, None)
       in
-      let exprList = List.map (fun (dotdotdot, e) -> match dotdotdot with
-      | Some dotdotdotLoc ->
-        (* If the `...` appears in any other location then the last,
-         * show a friendly, clear message explaining the consequences. *)
-        let msg = "Lists can only have one `...` spread, and at the end.
+      let msg = "Lists can only have one `...` spread, and at the end.
```

Comment:
Sorry could you change this one to remove `and`? The comma is probably enough to disambiguate meaning here.